### PR TITLE
Changed class User

### PR DESCRIPTION
**Pull request by Ai4n**
Hi

### CODE REVIEW

File: Server/src/main/ServerController.java [hunk 1]
```diff
@@ -175,21 +175,21 @@ private void searchBookInUserBooksList(int userId, String text) {
...
-	private void checkLogin(String newLogin) {
+	private void loginCheck(String newLogin) {
 		Boolean isLoginExist = userRepo.isLoginExist(newLogin);
 		IsLoginExistResponse isLoginExistResponse = new IsLoginExistResponse(isLoginExist);
 		socketController.write(isLoginExistResponse.json());
 	}
 
-	private void addUser(String login, String passWord) {
-		User user = new User(login, passWord, UserRole.USER);
+	private void addUser(int iDuser, String name, String surname, String login, String passWord, UserRole role) {
```
Comment (dnlbl) [inline Server/src/main/ServerController.java]: id

File: Server/src/repositories/UserRepo.java [hunk 2]
```diff
@@ -27,19 +27,23 @@ public UserRepo() {
...
+	public User userCheck(String login, String password) {
 		PreparedStatement statement;
+		
 		String sql = "SELECT * FROM users WHERE login = ? AND password = ?";
 		try {
 			statement = connection.prepareStatement(sql);
 			statement.setString(1, login);
 			statement.setString(2, password);
 			ResultSet rs = statement.executeQuery();
+			System.out.println(login);
```
Comment (dnlbl) [inline Server/src/repositories/UserRepo.java]: remove

File: Server/src/repositories/UserRepo.java [hunk 3]
```diff
@@ -27,19 +27,23 @@ public UserRepo() {
...
 			if (rs.next()) {
```
Comment (dnlbl) [inline Server/src/repositories/UserRepo.java:39]: n

File: Server/src/repositories/UserRepo.java [hunk 4]
```diff
@@ -27,19 +27,23 @@ public UserRepo() {
...
-				int idUser = rs.getInt(1);
-				String role = rs.getString(4);
-				UserRole userRole = UserRole.create(role);
-				User user = new User(idUser, login, password, userRole);
+				String name = rs.getString(2);
+				String surname = rs.getString(3);
+				String roleStr = rs.getString(6);
+				System.out.println(roleStr);
```
Comment (dnlbl) [inline Server/src/repositories/UserRepo.java]: remove

File: Server/src/repositories/UserRepo.java (hunk 2 above)
Comment (dnlbl) [inline Server/src/repositories/UserRepo.java]: remove

File: Server/src/main/ServerController.java (hunk 1 above)
Comment (dnlbl) [inline Server/src/main/ServerController.java]: id

File: Server/src/main/ServerController.java [hunk 5]
```diff
@@ -175,21 +175,21 @@ private void searchBookInUserBooksList(int userId, String text) {
 		socketController.write(searchInUserBooksResponse.json());
 	}
 
-	private void checkUser(String login, String password) {
-		User user = userRepo.checkUser(login, password);
+	private void userCheck(String login, String password) {
```
Comment (dnlbl) [inline Server/src/main/ServerController.java]: revert to checkUser

File: Server/src/main/ServerController.java [hunk 6]
```diff
@@ -175,21 +175,21 @@ private void searchBookInUserBooksList(int userId, String text) {
...
+		User user = userRepo.userCheck(login, password);
 		boolean isCredentialsCorrect = (user != null) ? true : false;
 		UserCheckResponse userCheckResponse = new UserCheckResponse(isCredentialsCorrect, user);
 		socketController.write(userCheckResponse.json());
 	}
 
-	private void checkLogin(String newLogin) {
+	private void loginCheck(String newLogin) {
```
Comment (dnlbl) [inline Server/src/main/ServerController.java]: revert to checkLogin

File: Server/src/repositories/UserRepo.java [hunk 7]
```diff
@@ -95,12 +101,13 @@ public void deleteUser(int userId) {
 			Statement statement = connection.createStatement();
 			ResultSet rs = statement.executeQuery(query);
 			while (rs.next()) {
-				int iduser = rs.getInt(1);
-				String login = rs.getString(2);
-				String password = rs.getString(3);
-				String userRole = rs.getString(4);
+				String name = rs.getString(2);
```
Comment (dnlbl) [inline Server/src/repositories/UserRepo.java:102]: add fetching of userId

File: Server/src/socketExchange/AddUserRequest.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
+package socketExchange;
+
+import com.google.gson.Gson;
+
+import main.ServerMessage;
+import main.UserRole;
+
+public class AddUserRequest extends SocketExchange{
+
+	private int iDuser;
```
Comment (dnlbl) [inline Server/src/socketExchange/AddUserRequest.java]: encapsulate data in User object

File: ServerClient/src/main/ClientController.java [hunk 9]
```diff
@@ -62,13 +69,13 @@ public void signUp() {
...
 			System.out.println("Repeat password (at least four simbols): ");
-			password2 = scan.next();
-		} while (!(password1.equals(password2)));
+			passwordTmp2 = scan.next();
+		} while (!(passwordTmp1.equals(passwordTmp2)));
 
-		cryptPassword = PasswordUtils.encodePassword(password1);
-		AddUserRequest addUserRequest = new AddUserRequest(newLogin, cryptPassword);
+		cryptPassword = PasswordUtils.encodePassword(passwordTmp1);
+		AddUserRequest addUserRequest = new AddUserRequest(0, name, surname, newLogin, cryptPassword, UserRole.USER);
```
Comment (dnlbl) [inline ServerClient/src/main/ClientController.java]: User user = new User(....)
AddUserRequest addUserRequest = new AddUserRequest(user);

File: ServerClient/src/main/ClientController.java [hunk 10]
```diff
@@ -78,8 +85,8 @@ public User login() {
 			System.out.println("Enter your login:");
 			String login = scan.next();
 			System.out.println("Enter your password:");
-			String password1 = scan.next();
-			String password = PasswordUtils.encodePassword(password1);
+			String passwordStr = scan.next();
```
Comment (dnlbl) [inline ServerClient/src/main/ClientController.java]: password

File: ServerClient/src/main/ClientController.java [hunk 11]
```diff
@@ -78,8 +85,8 @@ public User login() {
...
+			String password = PasswordUtils.encodePassword(passwordStr);
```
Comment (dnlbl) [inline ServerClient/src/main/ClientController.java]: hashedPassword

File: ServerClient/src/socketExchange/AddUserRequest.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
+package socketExchange;
+
+import com.google.gson.Gson;
+
+import main.ServerMessage;
+import main.UserRole;
+
+public class AddUserRequest extends SocketExchange{
+
+	private int iDuser;
```
Comment (dnlbl) [inline ServerClient/src/socketExchange/AddUserRequest.java]: encapsulate data in user class

**Review by dnlbl — CHANGES_REQUESTED** [review]
fix

File: Server/src/repositories/UserRepo.java (hunk 2 above)
Comment (Ai4n) [reply Server/src/repositories/UserRepo.java]: +

File: Server/src/main/ServerController.java (hunk 1 above)
Comment (Ai4n) [reply Server/src/main/ServerController.java]: +

File: Server/src/main/ServerController.java (hunk 5 above)
Comment (Ai4n) [reply Server/src/main/ServerController.java]: +

File: ServerClient/src/main/ClientController.java (hunk 9 above)
Comment (Ai4n) [reply ServerClient/src/main/ClientController.java]: +

File: ServerClient/src/main/ClientController.java (hunk 10 above)
Comment (Ai4n) [reply ServerClient/src/main/ClientController.java]: +

File: ServerClient/src/main/ClientController.java (hunk 11 above)
Comment (Ai4n) [reply ServerClient/src/main/ClientController.java]: +

File: ServerClient/src/socketExchange/AddUserRequest.java (hunk 12 above)
Comment (Ai4n) [reply ServerClient/src/socketExchange/AddUserRequest.java]: +

**Review by dnlbl — APPROVED** [review]

**Pull request merged**